src/__test__/AnomalyDetection/index.test.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { shallow } from 'enzyme';
+import AnomalyDetection from '../../AnomalyDetection/index';
+import QuerySyntax from '../../QuerySyntax/index';
+import NoContent from '../../NoContent/index';
+
+describe('<AnomalyDetection />', () => {
+  let wrapper;
+
+  const props = {
+    anomalyData: [
+      {
+        key_as_string: '2017-08-01T00:00:00.000-04:00',
+        matching_results: 10,
+        anomaly: 0.5,
+      },
+    ],
+    query: {
+      text: 'foo',
+    },
+  };
+
+  it('renders without crashing', () => {
+    const div = document.createElement('div');
+    ReactDOM.render(<AnomalyDetection {...props} />, div);
+  });
+
+  describe('formatDate', () => {
+    it('formats a date in MM/DD', () => {
+      const actual = AnomalyDetection.formatDate(props.anomalyData[0].key_as_string);
+
+      expect(actual).toEqual('08/01');
+    });
+  });
+
+  describe('when rendering charts', () => {
+    beforeEach(() => {
+      wrapper = shallow(<AnomalyDetection {...props} />);
+    });
+
+    describe('when dotProps does not have an anomaly', () => {
+      const dotProps = {
+        payload: {},
+      };
+
+      it('does not pass any color to the <Dot /> for renderDot', () => {
+        const actual = wrapper.instance().renderDot(dotProps).props;
+
+        expect(actual.fill).toBeUndefined();
+        expect(actual.stroke).toBeUndefined();
+      });
+
+      it('does not pass any color to the <Dot /> for renderActiveDot', () => {
+        const actual = wrapper.instance().renderActiveDot(dotProps).props;
+
+        expect(actual.fill).toBeUndefined();
+        expect(actual.stroke).toBeUndefined();
+      });
+    });
+
+    describe('when dotProps has an anomaly', () => {
+      const dotProps = {
+        payload: {
+          anomaly: 0.5,
+        },
+      };
+
+      it('passes the anomaly color to <Dot /> for renderDot', () => {
+        const actual = wrapper.instance().renderDot(dotProps).props;
+
+        expect(actual.fill).toEqual(AnomalyDetection.defaultProps.colorAnomaly);
+        expect(actual.stroke).toEqual(AnomalyDetection.defaultProps.colorAnomaly);
+      });
+
+      it('passes the active anomaly color to <Dot /> for renderActiveDot', () => {
+        const actual = wrapper.instance().renderActiveDot(dotProps).props;
+
+        expect(actual.fill).toEqual(AnomalyDetection.defaultProps.colorAnomalyActive);
+        expect(actual.stroke).toEqual(AnomalyDetection.defaultProps.colorAnomalyActive);
+      });
+    });
+
+    describe('when tooltipProps has an anomaly', () => {
+      const tooltipProps = {
+        payload: [
+          {
+            payload: {
+              anomaly: 0.5,
+            },
+          },
+        ],
```

Comment:
all of this comma-dangle. One of the few things I disliked about airbnb's linter.